### DISCUSSION TITLE
Should we add the option to switch to blocking request as well as async

### DISCUSSION THREAD

**Discussion by devinpearson** [walletd/walletd#75] [Ideas]
taken from: https://github.com/walletd/walletd/issues/61
https://tokio.rs/tokio/tutorial. Tokio docs mention under the section "When not to use Tokio"

Sending a single web request. The place where Tokio gives you an advantage is when you need to do many things at the same time. If you need to use a library intended for asynchronous Rust such as [reqwest](https://docs.rs/reqwest/), but you don't need to do a lot of things at once, you should prefer the blocking version of that library, as it will make your project simpler. Using Tokio will still work, of course, but provides no real advantage over the blocking API. If the library doesn't provide a blocking API, see [the chapter on bridging with sync code](https://tokio.rs/tokio/topics/bridging).

**Comment by karlbuys** [comment]
I'm thinking we should consider this a bit further down the line. As it stands right now, we can probably do better if we focus on new features. 

The biggest GOTCHA that may trip us up with blocking functionality is WASM-specific. WASM doesn't support blocking requests, so we would need to gate all specific blocking request functions behind a feature flag.

As long as we gate this feature, I think the option is useful to have, but I feel like there are other aspects to focus on that are more valuable.
